### ISSUE TITLE
`pane_search_string` does not work as expected with `-i` in `command-prompt`

### ISSUE THREAD

**Issue by alberti42** [tmux/tmux#4309]
# Bug Report: `pane_search_string` Does Not Work as Expected with `-i` in `command-prompt`

## **Description**
The `pane_search_string` variable is intended to provide the last search string in `copy-mode` and is used in tmux's default key binding for incremental search:

```
bind-key -T copy-mode C-r command-prompt -i -I "#{pane_search_string}" -T search -p "(search up)" { send-keys -X search-backward-incremental "%%" }
```

However, this functionality does not behave as expected. Specifically:

1. **When the `-i` flag is included**:
   - The `pane_search_string` is ignored or not pre-filled in the prompt.
   - The user starts with an empty prompt instead of the last search string.

2. **When the `-i` flag is omitted**:
   - The `pane_search_string` is pre-filled as expected.
   - However, the incremental search behavior (real-time updates) is lost.

This makes it impossible to use both incremental search (`-i`) and pre-fill the prompt with the previous search string (`-I "#{pane_search_string}"`).

## **Steps to Reproduce**
1. Open tmux and enter a session.
2. Enter `copy-mode` and perform a search with `/` or `?`.
3. Try triggering the default `C-r` binding for backward incremental search:
```
command-prompt -i -I "#{pane_search_string}" -T search -p "(search up)" { send-keys -X search-backward-incremental "%%" }
```
4. Observe the following behavior:
   - The prompt does not pre-fill with the last search string (`pane_search_string`).
   - Typing into the prompt works for incremental search, but the pre-filled string is missing.

## **Expected Behavior**
When the `-i` flag is used with `command-prompt`:
- The prompt should pre-fill with the value of `pane_search_string` provided by `-I`.
- Incremental search should work as expected, with real-time updates.

## **Actual Behavior**
- With `-i`, the `pane_search_string` variable is not pre-filled.
- Without `-i`, the `pane_search_string` is correctly pre-filled, but incremental search functionality is lost.

## **Environment**
- **tmux version**: 3.4 (installed via Homebrew)
- **Platform**: macOS 15.2 24C101 (arm64)
- **Host**: Mac14,7
- **Kernel**: 24.2.0
- **Packages**: 260 (brew)
- **Shell**: zsh 5.9
- **Terminal Emulator**: tmux
- **Resolution**: 1440x900
- **DE**: Aqua
- **WM**: Quartz Compositor
- **WM Theme**: Blue (Dark)
- **CPU**: Apple M2
- **GPU**: Apple M2
- **Memory**: 4542MiB / 24576MiB
- **XTERM**: screen-256color

## **Minimal Configuration to Reproduce**
```
bind-key -T copy-mode C-r command-prompt -i -I "#{pane_search_string}" -T search -p "(search up)" { send-keys -X search-backward-incremental "%%" }
```

## **Additional Notes**
- The issue seems tied to the interaction between `-i` (incremental search) and `-I` (initial input).
- Since this binding appears to be a default in tmux, it may affect users who rely on incremental search in `copy-mode`.

## **Suggested Fix**
Ensure that `-I "#{pane_search_string}"` pre-fills the prompt correctly when the `-i` flag is used in `command-prompt`.

**Comment by alberti42** [comment]
Ok. Wonderful and thank you for the explanation. I used `emacs` for many years, but not in recent times. So I forgot about it.

Essentially. When I restart a search, in my tmux with CTRL-R or CTRL-S in copy mode, the first time I press the key combination, the old search string is not shown. However, if I press the key sequence a second time, then the previous search string appears. This is wonderful and all what I needed!

Bottom line: I did not realize I need to press the key sequence a second time. I checked, this is indeed how `emacs` behaves. 

> If so, we could add a separate flag to make this work, but I don't see any reason to do that unless there is a use case.

I don't think a new flag is needed. Thank you for the amazing development!

PS: My recommendation would be to add a sentence in `man tmux` page. I read the page carefully, but I was unable to figure the right way to show the old search string. Do you want me to draft a possible sentence to add?

**Comment by nicm** [comment]
Yes if you want to suggest some text for the man page it would be great. `C-r` and `C-s` are a bit special case because even if they are bound to a different key you still need to press `C-r` or `C-s` to use the last search string... it is just meant to work like emacs.

**Comment by alberti42** [comment]
Cool. Yes, I did not mean to add my key bindings to the man page.

I made a suggestion below:

```
>>>OLD TEXT
The search commands come in several varieties: ‘search-forward’ and ‘search-backward’ search for a regular expression; the ‘-text’ variants
search for a plain text string rather than a regular expression; ‘-incremental’ perform an incremental search and expect to be used with the -i
flag to the command-prompt command.  ‘search-again’ repeats the last search and ‘search-reverse’ does the same but reverses the direction
(forward becomes backward and backward becomes forward).
<<<END OLD TEXT

For an incremental search, if you use the default key bindings,  

	bind-key -T copy-mode C-r command-prompt -i -I "#{pane_search_string}" -T search -p "(search up)" { send-keys -X search-back$ard-incremental "%%" }
	bind-key -T copy-mode C-s command-prompt -i -I "#{pane_search_string}" -T search -p "(search down)" { send-keys -X search-forward-incremental "%%" }

you can start it by pressing ‘C-r’ and ‘C-s’ key combinations in copy-mode. You can then use the key bindings associated with ‘search-again’
and ‘search-reverse’ to quickly move through the search results. If you want to edit the previous search string later, start a new search with
either ‘C-r’ or ‘C-s’, and repeat the key combination a second time in order to display the previous search string. This behavior mimics the
default behavior of emacs when searching for a string.
```

**Comment by nicm** [comment]
Man pages need to be written in neutral third person, so don't say "you". Also I don't like to quote the default key bindings because they are trivial to see and they might change. How about adding something like:

~~~~
The default incremental search key bindings,
.Ql C-r
and
.Ql C-s ,
are designed to emulate
.Xr emacs 1 .
When first pressed they allow a new search term to be entered; if pressed with an empty search
term they repeat the previously used search term.
